### ISSUE TITLE
TensorBoard Visualisation 

### ISSUE THREAD

**Issue by zfhall** [RyanZotti/Self-Driving-Car#115]
@RyanZotti First of all I apologise for how many questions I have had recently, I'm a real novice when it comes to Tensorflow and ML so I need all the help I can get! So thank you in advance if you are sparing the time to read this.

Currently I'm trying to figure out how to implement  accuracy and loss visualisation into your code. It seems you have already done this somewhat, however I cannot get it to work. All I have managed to do is visualise the weights in each layer and the model graph.  

For example in the code below you have created an operations to log a scalar, however I cannot see where you write the values into the log file later on? Am I missing something? 

```
    def train(self, sess, x, y_, accuracy, train_step, train_feed_dict, test_feed_dict):

        # To view graph: tensorboard --logdir=/Users/ryanzotti/Documents/repos/Self_Driving_RC_Car/tf_visual_data/runs
        tf.summary.scalar('accuracy_summary', accuracy)
        merged = tf.summary.merge_all()
```

**Comment by RyanZotti** [comment]
@zfhall Great question. Short answer is that my TensorBoard code is broken because I had started to dismantle it. I used to have working TensorBoard functionality, but back when I first started the project Google kept changing the TensorBoard API and it was a pain to keep the TensorBoard code updated. And on the other hand I felt there wasn't much to gain from TensorBoard. I never really found it that helpful, so now I just write my accuracy to a simple: [text file](https://github.com/RyanZotti/Self-Driving-Car/blob/master/Trainer.py#L169-L171): 
```
with open(self.results_file, 'a') as f:
    f.write(message.format(epoch, train_accuracy, test_accuracy)+'\n')
```
What do you want to use TensorBoard for in particular? It's possible Google added useful functionality to it since I last looked at it, but my working TensorBoard code is buried waaayyy back in my commit history, so it'll take me a while to find and test it.

**Comment by zfhall** [comment]
@RyanZotti That makes sense. Many of the tutorials I have looked at seem to use outdated code for TensorBoard and I have ran into many problems because of this. 

I wanted to use TensorBoard to visualise accuracies , and cross entropy losses. Mostly because I prefer things visualised and work with plots on a daily basis so its just the way I like to do things. I also don't plan on putting this project to bed any time soon so its a nice way to expand on what has already been done. 

Please don't go out of your way to go through your history though. I'm just very new to TensorBoard so with some more research I'm sure ill be able to figure it out.

**Comment by RyanZotti** [comment]
@zfhall If all you're trying to do is visualize the accuracy / loss, then writing to a file and using something like python's Seaborn [Seaborn](https://seaborn.pydata.org/) or [Matplotlib](https://matplotlib.org/) would probably be easier and less prone to changing code. Both libraries have been around much longer than TensorFlow and have much more stable code examples on Google.

**Comment by zfhall** [comment]
@RyanZotti Thanks for the info. I think ill just use matplotlib as i'm more familiar with it and your right it will be easier than this tensor board stuff!
